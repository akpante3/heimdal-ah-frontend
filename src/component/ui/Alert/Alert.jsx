import { toastr as feedback } from 'react-redux-toastr';

const Alert = ({ type, title, message }) => {
  if (type === 'warning') {
    return (feedback.warning(title, message));
  }
  return feedback.success(title, message);
};

export default Alert;
