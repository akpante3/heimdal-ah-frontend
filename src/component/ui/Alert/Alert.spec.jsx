/* eslint-disable no-undef */
import React from 'react';
import Alert from './Alert';

describe('Password Reset Component Test Suite', () => {
  describe('<Alert />', () => {
    it('should render the Alert component', () => {
      const wrapper = mount(<Alert type='success' title='Alert' message='This is alert' />);
      expect(wrapper.length).toBe(1);
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.props().type).toEqual('success');
      expect(wrapper.props().title).toEqual('Alert');
      expect(wrapper.props().message).toEqual('This is alert');
    });
  });
});
