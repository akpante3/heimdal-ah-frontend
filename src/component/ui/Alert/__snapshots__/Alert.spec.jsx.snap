// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Password Reset Component Test Suite <Alert /> should render the Alert component 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <Alert
    message="This is alert"
    title="Alert"
    type="success"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "message": "This is alert",
      "title": "Alert",
      "type": "success",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "message": "This is alert",
        "title": "Alert",
        "type": "success",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
